src/mlpack/methods/isomap/mds.hpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file mds.hpp
+ * @author Rishabh Ranjan
+ * 
+ * This file defines MDS class which implements classical Multidimensional
+ * Scaling which is a dimensionality reduction algorithm which uses spectral
+ * decomposition of dissimilarity matix.
+ * This implementation is based on MATLAB's implementation of classical
+ * multidimensional scaling (file - cmdscale.m).
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_ISOMAP_MDS
+#define MLPACK_METHODS_ISOMAP_MDS
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace isomap {
+
+/**
+ * This class implements classical Multidimensional Scaling.
+ */
+class MDS
+{
+ public:
+  /**
+   * Fucntion to implement classical Multidimensional Scaling on a given
+   * dissimilarity matrix.
+   * 
+   * @param disMat -dissimilarity matrix (distance matrix)
+   */
+  void Apply(arma::mat& disMat)
+  {
+    // check to see if the incoming matrix is valid for cMDS
+    IsValidMatrix(disMat);
+
+    size_t n = disMat.n_rows;
+    arma::mat temp(n, n);
+
+    // Spectral decomposition
+
+    arma::mat H = arma::eye<arma::mat>(n, n) - temp.fill((double) (1.0/n));
+    arma::mat B = H * (-0.5 * (disMat % disMat) ) * H;
+    arma::mat eigen_vecs;
+    arma::vec eigen_vals;
+
+    // guard eigen values against round-off errors
+    arma::eig_sym(eigen_vals, eigen_vecs, (B + B.t())/2);
```

Comment:
This is a clever little trick for stability, but honestly I think that it is a bit costly because the size of `B` may be quite large.  I also suspect that it should not be a huge issue, so I don't think we need to do it.

Reply:
Okay. I verified this on multiple data and you are right. Would remove the operation in next commit.